dongle_app/js/app.js

```diff
@@ -151,4 +151,23 @@ window.onload = function() {
       showQR(dongle);
     }
   );
+
+  function updateOnlineStatus(event) {
+    var ele = document.getElementById('not-on-line');
+    if(navigator.onLine){
+      ele.classList.remove('open');
+    }else{
+      ele.classList.add('open');
+    }
+  }
+
+  window.addEventListener('online',  updateOnlineStatus);
+  window.addEventListener('offline', updateOnlineStatus);
+
+  document.addEventListener("mozvisibilitychange", function() {
```

Comment:
Check as well that the document is `hidden`. More info in https://developer.mozilla.org/en-US/docs/Web/Guide/User_experience/Using_the_Page_Visibility_API